src/domain_bridge/wait_for_graph_events.hpp

```diff
@@ -257,7 +262,15 @@ class WaitForGraphEvents
     result_qos.qos.deadline(max_deadline);
     result_qos.qos.lifespan(max_lifespan);
 
-    return result_qos;
+    // If the number of endpoints has changed, get QoS again
```

Comment:
I would expect the time between here and the first time we query for endpoints above is negligible, so I'm not convinced this block is helping much.

Reply:
Fair enough, I've removed this in 119cf67